flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -428,6 +429,10 @@ public final void invoke() throws Exception {
 				// See FLINK-7430
 				isRunning = false;
 			}
+			MailboxExecutor mainMailboxExecutor = mailboxProcessor.getMainMailboxExecutor();
```

Comment:
I'd rather add a method drain to mailboxProcessor that does the following lines.